tasks/apply_catalog.rb

```diff
@@ -29,6 +29,10 @@
 # Make sure to apply the catalog
 Puppet[:noop] = args['_noop'] || false
 
+args['apply_settings'].each do |setting, value|
```

Comment:
If the param is optional, we need to handle it being unset here

Reply:
oh, yeah
 oops.